Ben expand inactivitymessages (#49)
fixie-ai/ultradox: docs/src/content/docs/api/calls.mdx

    <dd class="ml-4"><span class="text-gray-600">pattern</span> - `^-?(?:0|[1-9][0-9]{0,11})(?:\.[0-9]{1,9})?s$`</dd>
    <dd class="ml-4"><span class="text-gray-600">examples</span> - "60s", "5.5s"</dd>
  </div>
  <div>
    <dt class="font-mono">message</dt>
    <dd class="ml-4"><span class="text-gray-600">string</span> - The message to speak.</dd>
    <dd class="ml-4"><span class="text-gray-600">examples</span> - "Are you still there?", "If there's nothing else, I will end the call now."</dd>
  </div>
  <div>
    <dt class="font-mono">endBehavior</dt>
    <dd class="ml-4"><span class="text-gray-600">string</span> - The behavior to exhibit when the message is finished being spoken. Must be one of the enumerated values.</dd>
    <dd class="ml-4"><span class="text-gray-600">enum</span></dd>
    <dd class="ml-8">
      <ul class="list-disc space-y-2">
        <li><code>END_BEHAVIOR_UNSPECIFIED</code> - Default. The system will continue to wait for user input.</li>
        <li><code>END_BEHAVIOR_HANG_UP_SOFT</code> - Will end the call unless the user interacts while the agent is delivering the message.</li>

>         <li><code>END_BEHAVIOR_HANG_UP_STRICT</code> - Immediately terminates the call after speaking the message, without waiting for any additional user interaction.</li>

Reviewer: ```suggestion
        <li><code>END_BEHAVIOR_HANG_UP_STRICT</code> - Will end the call after speaking the message, regardless of whether the user interrupts.</li>
```

      </ul>
    </dd>
  </div>
</dl>

##### Example

```js
{
  "systemPrompt": "You are a helpful assistant.",
  "inactivityMessages": [
    {

>       "duration": "5s",
>       "message": "Hello, are you still there?"
>     },
>     {
>       "duration": "10s",
>       "message": "If there's nothing else, may I end the call?"
>     },
>     {
>       "duration": "5s",
>       "message": "Thank you for calling. Have a great day. Goodbye.",
>       "endBehavior": "END_BEHAVIOR_HANG_UP_STRICT"

Reviewer: These durations are too tight for an example. Also non-strict hangups are probably preferable in most cases.

```suggestion
      "duration": "30s",
      "message": "Are you still there?"
    },
    {
      "duration": "30s",
      "message": "If there's nothing else, may I end the call?"
    },
    {
      "duration": "45s",
      "message": "Thank you for calling. Have a great day. Goodbye.",
      "endBehavior": "END_BEHAVIOR_HANG_UP_SOFT"
```

Author: committing this but will edit durations a bit

    }
  ]
}
```

Here's what would happen based on the example above:

<Steps>

> 1. Call starts.
> 1. After 5 seconds of no user interaction, agent says "Hello, are you still there?".
> 1. If user interacts, call continues. If no user interaction occurs for 10 seconds, agent says "If there's nothing else, may I end the call?".
> 1. If no user interaction occurs for another 5 seconds, agent says the provided message and the call ends.

Reviewer: ```suggestion
1. Call starts.
1. After 30 seconds of no user interaction, agent says "Are you still there?".
1. If user interacts, call continues. If no user interaction occurs for another 30 seconds, agent says "If there's nothing else, may I end the call?".
1. If no user interaction occurs for another 45 seconds, agent says the provided message and the call ends unless the agent is interrupted during this final message.
```

```

#### Using Mistral
:::caution[Important Note]

> Mistral uses a simplified prompt structure.

Reviewer: ```suggestion
Mistral requires a simplified prompt structure.
```
